### PR TITLE
Add validations for range options

### LINKED ISSUE #20: Range option validation
- [ ] create custom validation in range option model 
When range options are created for range type questions, if the building operator's answer sits within a range specified by a range option, a conditional question might be triggered. For example:

**Admins question creation:**
Q: How many floors are in the building?
Range options:
5-10, 11-15, 16-20
Follow up Q's: 
If answer is in 5-10, ask: Is there an elevator?

**Building operator questionnaire form:**
Q: How many floors are in the building?
Building Operator answer: 7
Q: Is there an elevator?

**This means that range options for a single range question CANNOT have overlapping min-max ranges. For example:**

**Admins question creation:**
Q: How many floors are in the building?
Range options:
5-10, 4-10
If answer is in 5-10, ask: Is there an elevator?

If a building operator answers 7, it is ambiguous which range option is being triggered, which in turn makes it unclear whether or not to show the followup question that's attached to only one option.

### PR DESCRIPTION
**Pull request by kevintli**
Resolves #20 

Range options need to have valid min and max values that do not overlap
with any other ranges for the same question (otherwise, it would be
unclear which child question to trigger for certain answers).

### CODE REVIEW

File: app/models/range_option.rb [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+    end
+  end
+
+  def ranges_overlap(range1, range2)
+    # Even if min > max, we can still check for overlap assuming the user intended to swap the min and max values
+    min1, max1 = [range1.min, range1.max].minmax
+    min2, max2 = [range2.min, range2.max].minmax
+    return [min1, min2].max <= [max1, max2].min
+  end
+end
```
Comment (HacksForHugs) [inline app/models/range_option.rb]: newlineeeee

Comment (kevintli) [reply app/models/range_option.rb]: Oh do I just add a new line at the end of the file?

Comment (HacksForHugs) [reply app/models/range_option.rb]: yah
install a ruby linter package :)

Comment (kevintli) [reply app/models/range_option.rb]: Okay! Will do

**Pull request merged**